src/hotspot/share/gc/serial/genMarkSweep.cpp

```diff
@@ -112,6 +112,8 @@ void GenMarkSweep::invoke_at_safepoint(ReferenceProcessor* rp, bool clear_all_so
 
   deallocate_stacks();
 
+  MarkSweep::_string_dedup_requests->flush();
```

Comment:
Similarly, `flush` should be called inside `mark_sweep_phase1`, right after ref-processing.

Reply:
Thanks for your review and Kim's detailed explanation.

I think putting this invocation here or the position you suggest should have no effect on the result, so I tend not to make this modification.

Reply:
I believe it's desirable to narrow the request window (btw first `add` and `flush`), which offers more prompt string-dedup and memory release (in `flush`). Additionally, it makes more sense to call `flush` right after the complete live objects traversal  (strong-marking + final-marking as part of ref-processing), since we will not visit any new objects then.

Reply:
I prefer it where it is.  Rather than putting something between phantom
reference processing an oopstorage-based weak processing, I would prefer to
merge those two.  I don't remember if there's an RFE for that.

Reply:
I still think `flush` should be called sooner, e.g. inside `mark_sweep_phase1`, which is not in conflict with merging phantom-processing and weak root processing. Anyway, this is subjective; feel free to integrate the current version.